src/main/java/formflow/library/FileController.java

```diff
@@ -153,6 +164,9 @@ public ResponseEntity<?> upload(
 
       return ResponseEntity.status(HttpStatus.OK).contentType(MediaType.TEXT_PLAIN).body(newFileId.toString());
     } catch (Exception e) {
+      if (e instanceof ResponseStatusException) {
```

Comment:
@bseeger I don't like this, but it was the only way I could think of to ensure a 404 gets thrown since this gobbles up exceptions of all types 🤔

Reply:
Does it not work to change 
```java
} catch (Exception e }) {
```
to
```java
} catch ResponseStatusException e }) {
```

Reply:
I think the Exception e was meant as a sort of catch all for potential errors / issues with uploading to S3, although I'm not sure what the exact errors that would be thrown in that case are. I assume those would just get thrown as a 500 either way though?